core/src/main/java/com/github/jsonldjava/core/JsonLdApi.java

```diff
@@ -1968,48 +1977,61 @@ public boolean isWellFormedListNode() {
             }
 
             // 3.5)
+            final Map<String, List<Node>> nodes = new HashMap<>();
+
             for (final RDFDataset.Quad triple : graph) {
                 final String subject = triple.getSubject().getValue();
                 final String predicate = triple.getPredicate().getValue();
                 final RDFDataset.Node object = triple.getObject();
+                final List<Node> list = nodes.getOrDefault(subject, new ArrayList());
+                list.add(new Node(predicate, object));
+                nodes.put(subject, list);
+            }
+            for (final Map.Entry<String, List<Node>> nodeEntry : nodes.entrySet()) {
+              final String subject = nodeEntry.getKey();
+
+                for (final Node n : nodeEntry.getValue()) {
+                    final String predicate = n.predicate;
+                    final RDFDataset.Node object = n.object;
+
+                    // 3.5.1+3.5.2)
+                    NodeMapNode node;
+                    if (!nodeMap.containsKey(subject)) {
```

Comment:
This can also be switched to use ``computeIfAbsent`` to remove the if statement and reduce it to a single line (I know this is previous code, but would be good to fix it now that we are relying on Java-8, which wasn't the case originally)